src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -252,5 +260,65 @@ public void Dispose()
         {
             _deadlineTimer?.Dispose();
         }
+
+        internal string GetGrpcEncoding()
+        {
+            // In order of precedence:
+            // 1. Compression algorithm of request
+            // 2. Compression algorithm configured on server
+            return _grpcEncoding ?? ServiceOptions.DefaultCompressionAlgorithm;
+        }
+
+        internal bool GrpcAcceptEncodingMatch(string encoding)
```

Comment:
nit: it's immediately not obvious what this method checks (=that the grpc-encoding is contained in grpc-accept-encoding). Let's add a comment or rename.